src/sinks/aws_s3/config.rs

```diff
@@ -176,13 +178,24 @@ impl S3SinkConfig {
 
         // Configure our partitioning/batching.
         let batch_settings = self.batch.into_batcher_settings()?;
-        let key_prefix = self
+        let key_prefix: Template = self
             .key_prefix
             .as_ref()
             .cloned()
             .unwrap_or_else(|| DEFAULT_KEY_PREFIX.into())
             .try_into()?;
-        let partitioner = KeyPartitioner::new(key_prefix);
+        let ssekms_key_id: Option<Template> = if self.options.ssekms_key_id.is_none() {
+            None
+        } else {
+            self.options
+                .ssekms_key_id
+                .as_ref()
+                .cloned()
+                .map(Template::try_from)
+                .unwrap()
+                .ok()
+        };
```

Comment:
I think this is simpler, plus it returns any error raised whilst creating the `Template` rather than setting in to `None`.

```suggestion
   
        let ssekms_key_id = self
            .options
            .ssekms_key_id
            .as_ref()
            .map(|ssekms_key_id| Template::try_from(ssekms_key_id.as_str()))
            .transpose()?;

```

Reply:
Ah, yeah that's much better...the `transpose` was what I was missing before 💯